### PR TITLE
fix!: getting key for different MetaMask version

### PR DESCRIPTION
**Pull request by BeroBurny**

### CODE REVIEW

File: packages/snap/src/utils/version.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+export const getMetamaskVersion = async (
+  wallet: SnapProvider
+): Promise<string> =>
+  (await wallet.request({
+    method: "web3_clientVersion",
+    params: [],
+  })) as string;
+
+export const isNewerVersion = (
```
Comment (mpetrunic) [inline packages/snap/src/utils/version.ts:11]: you should probably write some unit tests for this method

Comment (BeroBurny) [reply packages/snap/src/utils/version.ts:11]: it's a common method in snaps, probably the best way is to create a library for these utils and cover all that there.

Comment (mpetrunic) [reply packages/snap/src/utils/version.ts:11]: Why don't we use: https://www.npmjs.com/package/semver?

Comment (BeroBurny) [reply packages/snap/src/utils/version.ts:11]: it can be used, but first, it needs to be checked if it will pass `SES`

**Comment by BeroBurny** [comment]
stale

**Pull request closed without merging**